### PR TITLE
Migrate CI to github Actions

### PR DESCRIPTION
**Pull request by mantissahz**
~`e2e-test` is not included because there are many errors in my local testing.~

We will have the step `make e2e-test` after the step `make ci`.

cc @innobead, @bk201, @derekbit, @FrankYang0529

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    # For multi-platform support
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Declare Branch
+      run: |
+        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> "$GITHUB_ENV"
+    - name: Login to Docker Hub
```
Comment (FrankYang0529) [inline .github/workflows/build.yml:65]: In rancher organization, we may need to read docker credential like this.

https://github.com/harvester/load-balancer-harvester/blob/0025da293f105cc500a27795249e4f952d80adcd/.github/workflows/template-build.yml#L19-L21

https://github.com/harvester/load-balancer-harvester/blob/0025da293f105cc500a27795249e4f952d80adcd/.github/workflows/template-build.yml#L35-L41

**Review by FrankYang0529 — APPROVED** [review]
Overall LGTM. For pushing image, we can merge first and fix it if it can't work.

**Comment by derekbit** [comment]
I will merge it after releasing v0.0.27.

**Comment by derekbit** [comment]
@mantissahz 
Can you build images for arm and amd64 platforms?

**Comment by mantissahz** [comment]
@derekbit,

I can build arm64 and amd64 images as https://hub.docker.com/repository/docker/jamesluhz/local-path-provisioner/tags.

Do we also need the arm image (it is built in Drone CI)?

**Comment by derekbit** [comment]
> @derekbit,
> 
> I can build arm64 and amd64 images as https://hub.docker.com/repository/docker/jamesluhz/local-path-provisioner/tags.
> 
> Do we also need the arm image (it is built in Drone CI)?

Yes, need `arm` as well.

**Comment by mantissahz** [comment]
https://hub.docker.com/repository/docker/jamesluhz/local-path-provisioner/tags
Images built with arch `arm64`, `arm`, `amd64`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    - name: Checkout Codes
+      uses: actions/checkout@v4
+
+    # Build binaries
+    - name: Run CI
+      run: make ci
+
+    # Run e2e test
+    #- name: Run E2e Test
+    #  run: make e2e-test
```
Comment (derekbit) [inline .github/workflows/build.yml:24]: We can add `make e2e-test` now. I will fix the flaky issue later.

Comment (mantissahz) [reply .github/workflows/build.yml]: Sure. Added.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mantissahz) [reply .github/workflows/build.yml:79]: Modified. Thanks

**Review by derekbit — APPROVED** [review]

**Pull request merged**